package_reloader.py

```diff
@@ -14,20 +15,37 @@
     reload_lock = Lock()
 
 
-def casedpath(path):
-    # path on Windows may not be properly cased
-    # https://github.com/randy3k/AutomaticPackageReloader/issues/10
-    r = glob(re.sub(r'([^:/\\])(?=[/\\]|$)', r'[\1]', path))
-    return r and r[0] or path
+if sys.platform.startswith("win"):
+    def realpath(path):
+        # path on Windows may not be properly cased
+        # https://github.com/randy3k/AutomaticPackageReloader/issues/10
+        r = glob(re.sub(r'([^:/\\])(?=[/\\]|$)', r'[\1]', os.path.realpath(path)))
+        return r and r[0] or path
+else:
+    def realpath(path):
+        return os.path.realpath(path)
 
 
 def relative_to_spp(path):
     spp = sublime.packages_path()
-    spp_real = casedpath(os.path.realpath(spp))
-    for p in [path, casedpath(os.path.realpath(path))]:
-        for sp in [spp, spp_real]:
+    spp_real = realpath(spp)
+    for p in {path, realpath(path)}:
+        for sp in {spp, spp_real}:
             if p.startswith(sp + os.sep):
                 return p[len(sp):]
+
+    if not sys.platform.startswith("win"):
+        # we try to follow symlink if the real file is not located in spp
+        for p in {path, realpath(path)}:
+            for d in os.listdir(spp):
+                subdir = os.path.join(spp, d)
+                subdir_real = realpath(subdir)
+                if not (os.path.islink(subdir) and os.path.isdir(subdir)):
+                    continue
+                for sd in {subdir, subdir_real}:
+                    if p.startswith(sd + os.sep):
+                        return os.sep + d + p[len(sd):]
```

Comment:
`relative_to_spp()` is only used to get the package name or check whether `path` is under SPP. May it would make sense to make it return the package name? Then we'd return just `d` in this line. On the line 35, it would become more complex, though. And the function may be called smth like `package_of()`.

Reply:
I forgot why I needed the relative path, but the only use now is to extract the name of the package.

Reply:
So may be just rename/refactor it accordingly then?